### PR TITLE
[SPL] SwiftUI-mimicking layout

### PR DESCRIPTION
**Pull request by watt**
This is the work myself and @n8chur have done so far in pairing. This is sort of a "checkpoint" merge, so that we can make more iterations on this while also bringing in the previous prototype on top of it.

### CODE REVIEW

File: BlueprintUI/Sources/Element/Element+Debug.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import Foundation
+
+
+extension Element {
+
+    @_spi(Debugging)
+    public func debugScope(_ scope: String) -> Element {
```
Comment (n8chur) [inline BlueprintUI/Sources/Element/Element+Debug.swift:7]: We should probably come back to this at some point—I don't know if we gave it enough though (maybe it's fine though?).

File: BlueprintUI/Sources/Element/ElementContent.swift [hunk 2]
```diff
@@ -45,8 +45,19 @@ public struct ElementContent {
...
-        // TODO: switch on layoutMode
-        storage.measure(in: constraint, environment: environment, cache: cache)
+        switch layoutMode {
+        case .standard:
+            return storage.measure(in: constraint, environment: environment, cache: cache)
+        case .singlePass:
+            return storage.sizeThatFits(
+                proposal: ProposedViewSize(constraint),
+                context: .init(
+                    // TODO: Hoist upward?
```
Comment (n8chur) [inline BlueprintUI/Sources/Element/ElementContent.swift:55]: Should we do this before we get too deep into performance optimizations? Might this have a significant impact?

File: BlueprintUI/Sources/Internal/LayoutResultNode.swift [hunk 3]
```diff
@@ -18,8 +18,34 @@ extension Element {
...
+        switch layoutMode {
+        case .standard:
+            return layout(layoutAttributes: LayoutAttributes(frame: frame), environment: environment)
+        case .singlePass:
+            return singlePassLayout(
+                proposal: .init(frame.size),
+                context: SPLayoutContext(
+                    attributes: LayoutAttributes(frame: frame),
+                    environment: environment,
+                    // TODO: Hoist up?
```
Comment (n8chur) [inline BlueprintUI/Sources/Internal/LayoutResultNode.swift:30]: Same here regarding potentially doing this before we get too deep into performance optimizations.

**Review by n8chur — APPROVED** [review]

File: BlueprintUI/Sources/Element/ElementContent.swift (hunk 2 above)
Comment (watt) [reply BlueprintUI/Sources/Element/ElementContent.swift:55]: Yeah, I'll make a note in our planning doc

**Pull request merged**